lib/vdsm/virt/vm.py

```diff
@@ -1183,6 +1183,26 @@ def _load_corrected_timeout(self, base, doubler=20, load=None):
             load = len(self.cif.vmContainer)
         return base * (doubler + load) / doubler
 
+    def _disk_preparation_timeout(self):
+        """
+        Return approximate timeout for disk preparation.
+
+        Some disk devices require modifying and reloading udev rules.
+        Currently, udev rules are reloaded for each such device individually.
+        This function returns a wild-guess timeout value based on the number
+        of such devices.
+
+        :returns: The timeout in seconds (int).
+        """
+        n_disks_to_appropriate = 0
+        for element in self._domain.devices.findall('./disk/source'):
```

Comment:
You can use:

    findall('./disk/source[@dev]')

to select all elements with a dev= arttibute. This can be more efficient since
the filtering is done in the C extension.

See https://docs.python.org/3/library/xml.etree.elementtree.html#supported-xpath-syntax